MergerCli/utils/BatchStatusManager.cs

```diff
@@ -43,23 +52,92 @@ public BatchStatusManager(string[] command)
             this.Command = command;
         }
 
-        public void SetCurrentBatch(string layer, string batchIdentifier)
+        public void SetCurrentBatch(string layer, string? batchIdentifier)
         {
-            this.States[layer].BatchIdentifier = batchIdentifier;
+            lock (_locker)
+            {
+                this.States[layer].BatchIdentifier = batchIdentifier;
+            }
+        }
+        
+        public ConcurrentDictionary<string, bool>? GetBatches(string layer)
+        {
+            lock (_locker)
+            {
+                if (this.States.ContainsKey(layer))
```

Comment:
Instead of this whole function i think you can write `return this.States[layer]?.Batches;.